internal/controller/components/modelregistry/modelregistry_controller_actions.go

```diff
@@ -147,6 +111,27 @@ func configureDependencies(ctx context.Context, rr *odhtypes.ReconciliationReque
 		return fmt.Errorf("failed to add default ingress secret for model registry: %w", err)
 	}
 
+	// To add SMM only when ServiceMesh is enabled
+	if _, err := cluster.GetCRD(ctx, rr.Client, ServiceMeshMemberCRD); err != nil {
```

Comment:
this seems to be similar to the check we were doing before as a gate, probably we should use the same machinery as setting the related condition ?  so the error would be more clear since the ServiceMesh operator should me manually installed.